wolfcrypt/src/wc_port.c

```diff
@@ -3162,6 +3162,32 @@ time_t xilinx_time(time_t * timer)
 
 time_t z_time(time_t * timer)
 {
+    /* Try to obtain the actual time from an RTC */
+    if (IS_ENABLED(CONFIG_RTC))
+	{
+		static const struct device *rtc = DEVICE_DT_GET(DT_NODELABEL(rtc));
+
+		if (device_is_ready(rtc))
+		{
+			struct rtc_time rtc_time;
+			struct tm *tm_time = rtc_time_to_tm(&rtc_time);
+
+			int ret = rtc_get_time(rtc, &rtc_time);
+
+			if (ret == 0)
+			{
+				time_t epochTime = mktime(tm_time);
+
+				if (timer != NULL)
+					*timer = epochTime;
+
+				return epochTime;
+			}
+		}
+	}
+
+	/* Fallback to uptime since boot. This works for relative times, but
+	 * not for ASN.1 date validation */
     struct timespec ts;
```

Comment:
Please avoid mixed declaration, so move `struct timespec ts;` to top of function. Please use `if () {` on same line.

Reply:
updated the commit, thanks